scripts/go-dep.mk

```diff
@@ -0,0 +1,127 @@
+# Default configuration
+IMG_NAME := $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)/$(APP_NAME)
+TAG := $(DOCKER_TAG)
+BASE_PKG := github.com/kyma-project/kyma
+IMG_GOPATH := /workspace/go
+IMG_GOCACHE := /root/.cache/go-build
+VERIFY_IGNORE := /vendor\|/automock
+
+# Other variables
+LOCAL_DIR = $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+COMPONENT_DIR = $(shell pwd)
+WORKSPACE_LOCAL_DIR = $(IMG_GOPATH)/src/$(BASE_PKG)/scripts
+WORKSPACE_COMPONENT_DIR = $(IMG_GOPATH)/src/$(BASE_PKG)/$(APP_PATH)
+FILES_TO_CHECK = find . -type f -name "*.go" | grep -v "$(VERIFY_IGNORE)"
+DIRS_TO_CHECK = go list ./... | grep -v "$(VERIFY_IGNORE)"
+DIRS_TO_IGNORE = go list ./... | grep "$(VERIFY_IGNORE)"
+
+# Base docker configuration
+DOCKER_CREATE_OPTS := -v $(LOCAL_DIR):$(WORKSPACE_LOCAL_DIR):delegated --rm -w $(WORKSPACE_COMPONENT_DIR) $(BUILDPACK)
+
+# Check if go is available
+ifneq (,$(shell go version 2>/dev/null))
+DOCKER_CREATE_OPTS := -v $(shell go env GOCACHE):$(IMG_GOCACHE):delegated -v $(shell go env GOPATH)/pkg/dep:$(IMG_GOPATH)/pkg/dep:delegated $(DOCKER_CREATE_OPTS)
+endif
+
+# Check if running with TTY
+ifeq (1, $(shell [ -t 0 ] && echo 1))
+DOCKER_INTERACTIVE := -i
+DOCKER_CREATE_OPTS := -t $(DOCKER_CREATE_OPTS)
+endif
+
+# Buildpack directives
+define buildpack-mount
+.PHONY: $(1)-local $(1)
+$(1):
+	@echo make $(1)
+	@docker run $(DOCKER_INTERACTIVE) \
+		-v $(COMPONENT_DIR):$(WORKSPACE_COMPONENT_DIR):delegated \
+		$(DOCKER_CREATE_OPTS) make $(1)-local
+endef
+
+define buildpack-cp-ro
+.PHONY: $(1)-local $(1)
```

Comment:
Should it be duplicated with `buildpack-mount` ?

Reply:
Yes, they shouldn't be sued togheter